synthesizer/src/vm/helpers/sequential_op.rs

```diff
@@ -138,3 +155,46 @@ pub enum SequentialOperationResult<N: Network> {
         )>,
     ),
 }
+
+#[cfg(feature = "announce-blocks")]
+fn start_block_announcement_stream() -> Option<Stream> {
+    let path = std::env::var("BLOCK_ANNOUNCE_PATH")
+        .map_err(|_| {
+            warn!("BLOCK_ANNOUNCE_PATH env variable must be set in order to publish blocks via IPC");
+        })
+        .ok()?
+        .to_fs_name::<local_socket::GenericFilePath>()
+        .expect("Invalid path provided as the BLOCK_ANNOUNCE_PATH");
+
+    match Stream::connect(path) {
+        Ok(stream) => {
+            debug!("Successfully (re)started the IPC stream for block announcements");
+            Some(stream)
+        }
+        Err(e) => {
+            warn!("Couldn't (re)start the IPC stream for block announcements: {e}");
+            None
+        }
+    }
+}
+
+#[cfg(feature = "announce-blocks")]
+fn announce_block(
+    stream: &mut Option<Stream>,
+    payload: (u32, Result<Vec<u8>, Box<bincode::ErrorKind>>),
+) -> Result<bool> {
+    if let Some(stream) = stream {
+        let (block_height, serialized_block) = payload;
+        let block_bytes = serialized_block?;
+        debug!("Announcing block {block_height} to the IPC stream");
+        let payload_size = u32::try_from(4 + block_bytes.len()).unwrap(); // Safe - blocks are smaller than 4GiB.
```

Comment:
nit: avoid magic number here

```suggestion
        let payload_size = u32::try_from(std::mem::size_of::<u32>() + block_bytes.len()).unwrap()); // Safe - blocks are smaller than 4GiB.
```

Reply:
Also, why do we panic here if the payload here is too large, but not, for example, if serialization of the block failed?

Reply:
I don't think a 4GiB block is possible at all